File: backend/src/models/Promo.js
const mongoose = require('mongoose');

const PromoSchema = new mongoose.Schema(
  {
    title: { type: String, required: true, trim: true },
    subtitle: { type: String, default: '' },
    badge: { type: String, default: '' },
    imageUrl: { type: String, default: '' },
    discountPercent: { type: Number, default: 0 },
    isActive: { type: Boolean, default: true }
  },
  { timestamps: true }
);

module.exports = mongoose.model('Promo', PromoSchema);

File: frontend/lib/models/promo.dart
class Promo {
  Promo({
    required this.id,
    required this.title,
    required this.subtitle,
    required this.badge,
    required this.imageUrl,
    required this.discountPercent,
  });

  final String id;
  final String title;
  final String subtitle;
  final String badge;
  final String imageUrl;
  final int discountPercent;

  factory Promo.fromJson(Map<String, dynamic> json) {
    return Promo(
      id: json['_id'] as String,
      title: json['title'] as String,
      subtitle: (json['subtitle'] ?? '') as String,
      badge: (json['badge'] ?? '') as String,
      imageUrl: (json['imageUrl'] ?? '') as String,
      discountPercent: (json['discountPercent'] ?? 0) as int,
    );
  }
}
